ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParsePayload.java

```diff
@@ -204,14 +208,17 @@ private void addAttributesFromPayload(Map<String, String> attributes, ObjectNode
           .ifPresent(v -> attributes.put(Attribute.OS_VERSION, v));
     }
 
-    // Try extracting variants of top-level client id.
-    Stream
-        .of(attributes.get(Attribute.CLIENT_ID), json.path(Attribute.CLIENT_ID).textValue(),
-            json.path("clientId").textValue())
-        .map(ParsePayload::normalizeUuid) //
-        .filter(Objects::nonNull) //
-        .findFirst() //
-        .ifPresent(v -> attributes.put(Attribute.CLIENT_ID, v));
+    if (Objects.isNull(attributes.get(Attribute.CLIENT_ID))) {
+      // Try extracting variants of top-level client id and normalize UUID in JSON payload.
+      Stream.of(Attribute.CLIENT_ID, "clientId") //
+          .filter(v -> Objects.nonNull(ParsePayload.normalizeUuid(json.path(v).textValue()))) //
+          .findFirst() //
+          .ifPresent(v -> {
+            final String normalizedUuid = ParsePayload.normalizeUuid(json.path(v).textValue());
+            json.put(v, normalizedUuid);
+            attributes.put(Attribute.CLIENT_ID, normalizedUuid);
+          });
+    }
```

Comment:
I think this block has gotten complex enough that it;s hard to parse through what's going on in the set of stream operations. And we have to call `ParsePayload.normalizeUuid(json.path(v).textValue())` twice on the same data.

I think I'd prefer that we have separate blocks at this point for CLIENT_ID and clientId. Or maybe it's worth  factoring out a function that accepts an `ObjectNode` and a string for the target key within it.

I am also okay if you want to make the judgment call that what's already here is the best option.

Reply:
I agree, having them in separate blocks makes it easier to see what's going on.